### PR TITLE
Image CDN: URL encode path parts

### PR DESCRIPTION
**Pull request by haqadn**
Related to #8321

## Proposed changes:

* `rawurlencode` each part of an image path while generating photon URL

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
No

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
None

### CODE REVIEW

File: projects/packages/image-cdn/src/class-image-cdn-core.php [hunk 1]
```diff
@@ -245,6 +244,22 @@ public static function cdn_url( $image_url, $args = array(), $scheme = null ) {
...
+	 * Input: "foo/bar baz/baz"
+	 * Output: "foo/bar%20baz/baz"
+	 *
+	 * @param string $path The path to escape.
+	 * @return string The escaped path.
+	 */
+	private static function escape_path( $path ) {
+		$parts = explode( '/', $path );
+		$parts = array_map( 'rawurlencode', $parts );
+		return implode( '/', $parts );
```
Comment (jeherve) [inline projects/packages/image-cdn/src/class-image-cdn-core.php:277]: Could we rely on `urlencode_deep()` here maybe, to simplify things?

**Review by jeherve — DISMISSED** [review]
This tests well for me, it doesn't seem to break anything. I wasn't able to reproduce the original issue though, so I cannot confirm that this will be enough to close the related issue.

File: projects/packages/image-cdn/src/class-image-cdn-core.php (hunk 1 above)
Comment (haqadn) [reply projects/packages/image-cdn/src/class-image-cdn-core.php:277]: `urlencode` converts spaces to `+` which doesn't work with path parts in photon https://i0.wp.com/boost.in.ngrok.io/wp-content/uploads/2024/09/Screenshot+2024-09-16-at-12.46.59%E2%80%AFPM.png

The purpose is served by `rawurlencode` https://i0.wp.com/boost.in.ngrok.io/wp-content/uploads/2024/09/Screenshot%202024-09-16-at-12.46.59%E2%80%AFPM.png

**Review by dilirity — APPROVED** [review]
Works as advertised.

:shipit:

**Pull request merged**